packages/core-flows/src/sales-channel/steps/associate-products-with-channels.ts

```diff
@@ -0,0 +1,47 @@
+import { Modules } from "@medusajs/modules-sdk"
+import { ContainerRegistrationKeys } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  links: {
+    sales_channel_id: string
+    product_ids: string[]
+  }[]
+}
+
+export const associateProductsWithSalesChannelsStepId =
+  "associate-products-with-channels"
+export const associateProductsWithSalesChannelsStep = createStep(
+  associateProductsWithSalesChannelsStepId,
+  async (input: StepInput, { container }) => {
+    const remoteLink = container.resolve(ContainerRegistrationKeys.REMOTE_LINK)
+
+    const links = input.links
```

Comment:
discussion: I believe that we are missing a validation step somewhere, unless we decide to trust the data and if an id does not exists it is the user problem. But currently, you could insert any id and it would work without issues. Do you think we should do something about that?